solver/llbsolver/bridge.go

```diff
@@ -43,56 +45,49 @@ func (b *llbBridge) Solve(ctx context.Context, req frontend.SolveRequest) (res *
 	if err != nil {
 		return nil, err
 	}
-	var cms []solver.CacheManager
-	for _, im := range req.CacheImports {
-		b.cmsMu.Lock()
-		var cm solver.CacheManager
-		cmId := identity.NewID()
-		if im.Type == "registry" {
-			// For compatibility with < v0.4.0
-			if ref := im.Attrs["ref"]; ref != "" {
-				cmId = ref
-			}
-		}
-		if prevCm, ok := b.cms[cmId]; !ok {
-			func(cmId string, im gw.CacheOptionsEntry) {
-				cm = newLazyCacheManager(cmId, func() (solver.CacheManager, error) {
-					var cmNew solver.CacheManager
-					if err := inVertexContext(b.builder.Context(ctx), "importing cache manifest from "+cmId, "", func(ctx context.Context) error {
-						resolveCI, ok := b.resolveCacheImporterFuncs[im.Type]
-						if !ok {
-							return errors.Errorf("unknown cache importer: %s", im.Type)
-						}
-						ci, desc, err := resolveCI(ctx, im.Attrs)
-						if err != nil {
-							return err
-						}
-						cmNew, err = ci.Resolve(ctx, desc, cmId, w)
-						return err
-					}); err != nil {
-						logrus.Debugf("error while importing cache manifest from cmId=%s: %v", cmId, err)
-						return nil, err
-					}
-					return cmNew, nil
-				})
-			}(cmId, im)
-			b.cms[cmId] = cm
-		} else {
-			cm = prevCm
-		}
-		cms = append(cms, cm)
-		b.cmsMu.Unlock()
-	}
 
-	if req.Definition != nil && req.Definition.Def != nil && req.Frontend != "" {
-		return nil, errors.New("cannot solve with both Definition and Frontend specified")
-	}
-
-	if req.Definition != nil && req.Definition.Def != nil {
+	loadResult := func(ctx context.Context) (solver.CachedResult, error) {
```

Comment:
Can we move this into a method on `(b *llbBridge) loadResult(cacheImports []client.CacheOptionsEntry) func (context.Context) (solver.CachedResult, error)`? It looks like only `req` is used from its closure.